ktor-server/ktor-server-core/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -53,11 +44,22 @@ class ContentNegotiation(val converters: List<ContentConverterRegistration>) {
             pipeline.sendPipeline.intercept(ApplicationSendPipeline.Render) {
                 if (subject is OutgoingContent) return@intercept
 
-                val suitableConverters = acceptedTypes().mapNotNull { (contentType, _) ->
+                val suitableConverters = call.request.acceptItems().mapNotNull { (contentType, _) ->
                     feature.converters.firstOrNull { it.contentType.match(contentType) }
                 }
 
-                val converted = suitableConverters.mapNotNull { it.converter.convertForSend(this, subject) }.firstOrNull()
+                val charset = call.request.acceptCharsetItems().mapNotNull { (charset, _) ->
```

Comment:
The main issue here is that `charset` is only relevant for text-producing content converters (plain-text, json) but it could be completely irrelevant for others (for example Java Serialization). Parsing, sorting and enumerating character encodings is releatively expensive and most likely need to be done in content converters rather than here